File: Week_4/problem_3-Shape_3.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219432/my

n = int(input())
s = n - 1         # Spaces before the first '*'
t = 1             # Number of '*' to print on the first row

for i in range(1, 2 * n + 1):
    print(" " * s, end="")  # Print leading spaces
    print("*" * t)          # Print stars

    # Update spaces and stars based on current row
    if i < n:
        s -= 1
        t += 2
    elif i > n:
        s += 1
        t -= 2

File: Week_4/problem_3-Shape_3.c
// https://codeforces.com/group/MWSDmqGsZm/contest/219432/my

#include <stdio.h>
#include <limits.h>
#include <string.h>
#include <math.h>

int main()
{
    int n;
    scanf("%d", &n);
    int s = n - 1;
    int t = 1;

    for (int i = 1; i <= 2 * n; i++)
    {
        for (int j = 1; j <= s; j++)
        {
            printf(" ");
        }
        for (int k = 1; k <= t; k++)
        {
            printf("*");
        }

        if (i < n)
        {
            s--;
            t += 2;
        }
        else if (i > n)
        {
            s++;
            t -= 2;
        }
        printf("\n");
    }

    return 0;
}